src/sklearn_raster/features.py

```diff
@@ -214,19 +222,23 @@ def _postprocess_ufunc_output(
         *,
         nodata_output: float | int,
         output_coords: dict[str, list[str | int]] | None = None,
+        set_attrs: dict[str, Any] | None = None,
     ) -> xr.DataArray:
         """Process the ufunc output by assigning coordinates and transposing."""
         if output_coords is not None:
             result = result.assign_coords(output_coords)
 
         # Transpose features from the last to the first dimension
         result = result.transpose(result.dims[-1], ...)
-
+        # Drop top-level attributes from the input data, but retain coordinate attrs to
+        # preserve the spatial reference, if present. Note that we avoid using
+        # drop_attrs(deep=False) for backwards compatibility due to
+        # https://github.com/pydata/xarray/issues/10027
+        result.attrs = {}
+        if set_attrs is not None:
+            result.attrs.update(set_attrs)
```

Comment:
This demonstrates my lack of familiarity with `xarray`, but it sound like the coordinate attrs are associated with coordinate objects whereas "regular" attrs would be the ones that get dropped here, is that right?